kos/views.py

```diff
@@ -154,19 +153,18 @@ def get(self, request, *args, **kwargs):
         return FileResponse(puzzle.file)
 
 
-class PuzzleSolutionView(UserPassesTestMixin, GetTeamMixin, DetailView):
+class PuzzleSolutionView(UserPassesTestMixin, DetailView):
     """Vráti PDF so šifrou"""
     model = Puzzle
 
     def test_func(self):
         puzzle = self.get_object()
         puzzle_year = puzzle.game.year
-        team = self.get_team()
-        team_year = None
+        team = self.request.user.team if self.request.user.is_authenticated and hasattr(
```

Comment:
Rozmyslam ze teraz mame tu logiku nakopirovanu z toho mixinu ... jeho problem je len ze dedi od `LoginRequiredMixin` ... Možno by nemusel a potom by mohla existovat jedna classa iba z `get_team` nazvana `GetTeamMixin` a potom `GetLoggedTeamMixin` ... alebo volaco take

Reply:
No hej ako nebol som ani ja prilis nadseny z toho riesenia, ale neviem premysliet uplne ako to je dobre ... ked mas dajaku predstavu tak si ochotny to prerobit?